README.md

```diff
@@ -105,38 +103,63 @@ React.render(<App person={person} />, document.body)
 person.name = "Mike" // will cause the Observer region to re-render
 ```
 
-Example2:
+In case you are a fan of render props, you can use that instead of children. Be advised, that you cannot use both approaches at once, children have a precedence.
+
+Here goes example showing render prop only
 
 ```javascript
-class App extends React.Component {
-  render() {
-     return (
-          <Provider h="hello" w="world">
-            <Comp />
-        </Provider>
-     )
-  }
-}
+
 const Comp = () => (
     <div>
         <Observer
-            inject={store => ({ h: store.h, w: store.w })}
-            render={props => <span>{`${props.h} ${props.w}`}</span>}
+            render={() => <span>hello world</span>}
```

Comment:
Hm, not very good example since there is nothing to observe :D Perhaps pass some variable in props to Comp and use that in render so it makes more sense.